ScanCommand/src/main/java/io/ballerina/scan/utils/ScanUtils.java

```diff
@@ -120,6 +130,124 @@ private static Target getTargetPath(Project project, String directoryName) {
         }
     }
 
+    /**
+     * Returns the {@link Path} of the html analysis report where generated issues are saved.
+     *
+     * @param issues        generated issues
+     * @param project       Ballerina project
+     * @param directoryName target directory name
+     * @return path of the html analysis report where generated issues are saved
+     */
+    public static Path generateScanReport(List<Issue> issues, Project project, String directoryName) {
+        JsonObject jsonScannedProject = new JsonObject();
+        jsonScannedProject.addProperty("projectName", project.currentPackage().packageName().toString());
+        Map<String, JsonObject> jsonScanReportPathAndFile = new HashMap<>();
+        issues.forEach((issue) -> {
+            IssueImpl issueImpl = (IssueImpl) issue;
+            String filePath = issueImpl.filePath();
+
+            JsonObject jsonScanReportFile;
+            if (!jsonScanReportPathAndFile.containsKey(filePath)) {
+                jsonScanReportFile = new JsonObject();
+                jsonScanReportFile.addProperty("fileName", issueImpl.fileName());
+                jsonScanReportFile.addProperty("filePath", filePath);
+                String fileContent = "";
+                try {
+                    fileContent = Files.readString(Path.of(filePath));
+                } catch (IOException ex) {
+                    throw new RuntimeException(ex);
```

Comment:
Let's append a message to the exception message to make it more context-sensitive. 
```suggestion
                    throw new RuntimeException("Failed to read the file with exception: " + ex.getMessage());
```

Reply:
Follow the same approach for other places as well.